projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -1314,14 +1318,36 @@ public void saveSecretData(String name, Map<String, byte[]> data) throws ApiExce
         try {
             nsSecret = api.replaceNamespacedSecret(name, this.namespace, secret, null, null, null);
         } catch (ApiException e) {
+            log.warn("Error while trying to save K8S secret", e);
             if (e.getCode() == 404) {
-                log.debug("Secret {} does not exits. Creating ...");
+                log.debug("Secret does not exist. Creating ...");
                 nsSecret = api.createNamespacedSecret(this.namespace, secret, null, null, null);
             } else {
+                log.error("Failed to save k8s secret: {}" , name);
                 throw e;
             }
         }
         log.debug("Saved k8s secret: {}", name);
+        logSecretDebugInformation(nsSecret);
+    }
+
+    /**
+     * This method logs secret information for debugging purposes.
+     * While also omitting any sensitive data. We don't want to throw
+     * any exceptions from this method, since it is only meant for logging.
+     *
+     * @param nsSecret
+     */
+    private void logSecretDebugInformation(V1Secret nsSecret) {
+        try {
+            var dataKeys = Optional.ofNullable(nsSecret.getData()).map(secret -> secret.keySet()).orElse(null);
+            var metaData = Optional.ofNullable(nsSecret.getMetadata()).map(secret -> secret.toString()).orElse(null);
+            var stringDataKeys = Optional.ofNullable(nsSecret.getStringData()).map(secret -> secret.keySet()).orElse(null);
+            log.debug("Replaced namespaced secret. Data keys : {}, MetaData: {}, StringData keys: {}, Type: {}, ApiVer: {}",
```

Comment:
secret name?

Reply:
It's already logged here. https://github.com/vmware/versatile-data-kit/blob/894bfcbc30259d11d07b25022831d54aa590b259/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java#L1330